src/sentry/integrations/vercel/webhook.py

```diff
@@ -142,27 +143,53 @@ def post(self, request: Request) -> Response:
 
         # Vercel's webhook allows you to subscribe to different events,
         # denoted by the `type` attribute. We currently subscribe to:
-        #     * integration-configuration-removed (Configuration Removed)
-        #     * deployment (Deployment Created)
+        #     * integration-configuration-removed, integration-configuration.removed (Configuration Removed)
+        #     * deployment, deployment.created (Deployment Created)
         # https://vercel.com/docs/integrations/webhooks-overview
-        try:
-            event_type = request.data["type"]
-        except KeyError:
-            return self.respond({"detail": "Missing event type."}, status=400)
+        with configure_scope() as scope:
+            try:
+                event_type = request.data["type"]
+            except KeyError:
+                return self.respond({"detail": "Missing event type."}, status=400)
+
+            # Try the new Vercel request. If it fails, try the old Vercel request
+            try:
+                payload = request.data["payload"]
+                external_id = (
+                    payload.get("team")["id"] if payload.get("team") else payload["user"]["id"]
+                )
+                scope.set_tag("vercel_webhook.type", "new")
 
-        external_id = request.data.get("teamId") or request.data["userId"]
+                if event_type == "integration-configuration.removed":
+                    configuration_id = request.data["payload"]["configuration"]["id"]
+                    return self._delete(external_id, configuration_id, request)
+                if event_type == "deployment.created":
+                    return self._deployment_created(external_id, request)
+            except KeyError:
```

Comment:
This works checking for the KeyError but with `.get` accessors we may end up with `None` for the ext_id or conf_id and fail with some other error in `self._delete` or `self._deployment_created`.

It may be worth refactoring to ensure no KeyErrors with safe `.get`'s, and just check if the id is none before trying the legacy pattern

Some pseudocode:

```
external_id = ... #some safe .get() on new payload
if external_id is None:
   external_id = # legacy payload
   
conf_id = ... #some safe .get() on new payload
if conf_id is None:
   conf_id = # legacy payload
   
if event_type in ['integration-configuration.removed', "integration-configuration-removed"]
  # remove
if event_type in ['deployment', 'deployment.created']
  # create deploy
```

Reply:
This is just a suggestion, if you want to stick with the try/except approach that's totally fine as well, but you need to ensure a KeyError, or use a more generic Exception.

Reply:
Thanks for the suggestion!

I decided to keep the try/except just so the code pertaining to the old payload can be removed more easily in the future, but I implemented your suggested to make it a more generic Exception :)